### PR TITLE
feat: busiest day of the week graph

### PR DESCRIPTION
**Pull request by nadav-pesach**

### CODE REVIEW

File: app/templates/agenda.html [hunk 1]
```diff
@@ -25,28 +19,45 @@
...
-        {% endfor %}
-    {% endfor %}
+        {% endfor %} {% endfor %}
+    </div>
+
+
+    <div class="col-4">
+        <div class="card-body">
+            <canvas id="myChart" width="400" height="400"></canvas>
+            <script src="../static/graph.js"></script>
```
Comment (yammesicka) [inline app/templates/agenda.html]: Prefer loading all the `script`s in the end of the `body` tag

File: app/static/graph.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+function makeGraph(events) {
```
Comment (yammesicka) [inline app/static/graph.js]: Which graph? Please change the names in the code to be more indicative :)

File: app/routers/agenda.py [hunk 3]
```diff
@@ -39,6 +40,18 @@ def agenda(
     """Route for the agenda page, using dates range or exact amount of days."""
 
     user_id = 1  # there is no user session yet, so I use user id- 1.
+
+    start_date_graph, end_date_graph = calc_dates_range_for_agenda(0, 0, 7)
```
Comment (yammesicka) [inline app/routers/agenda.py]: Why 7? Try to avoid magic numbers :)

File: app/routers/agenda.py [hunk 4]
```diff
@@ -39,6 +40,18 @@ def agenda(
...
+
+    events_objects_for_graph = agenda_events.get_events_per_dates(
+        db, user_id, start_date_graph, end_date_graph
+    )
+    events_for_graph = {
+        str(start_date_graph + timedelta(i)): 0 for i in range(8)}
```
Comment (yammesicka) [inline app/routers/agenda.py]: Why 8?

File: app/routers/agenda.py [hunk 5]
```diff
@@ -39,6 +40,18 @@ def agenda(
     """Route for the agenda page, using dates range or exact amount of days."""
 
     user_id = 1  # there is no user session yet, so I use user id- 1.
+
+    start_date_graph, end_date_graph = calc_dates_range_for_agenda(0, 0, 7)
+
+    events_objects_for_graph = agenda_events.get_events_per_dates(
```
Comment (yammesicka) [inline app/routers/agenda.py]: The name of the variables here can be a bit better :)

**Review by yammesicka — COMMENTED** [review]
Great start! I've added few insights. Please take a look :)

File: app/templates/agenda.html (hunk 1 above)
Comment (nadav-pesach) [reply app/templates/agenda.html]: done!
and also changed to:    
<script src="{{ url_for('static', path='/graph.js') }}"></script>

File: app/static/graph.js (hunk 2 above)
Comment (nadav-pesach) [reply app/static/graph.js]: done

File: app/routers/agenda.py (hunk 3 above)
Comment (nadav-pesach) [reply app/routers/agenda.py]: done

File: app/routers/agenda.py (hunk 4 above)
Comment (nadav-pesach) [reply app/routers/agenda.py]: well when it was 7 it actually returned 8 days
I changed it to return only a week

File: app/routers/agenda.py (hunk 5 above)
Comment (nadav-pesach) [reply app/routers/agenda.py]: done

File: app/routers/agenda.py [hunk 6]
```diff
@@ -39,6 +40,21 @@ def agenda(
     """Route for the agenda page, using dates range or exact amount of days."""
 
     user_id = 1  # there is no user session yet, so I use user id- 1.
+
+    get_a_week_int = 6
```
Comment (yammesicka) [inline app/routers/agenda.py]: why the `get` prefix? Also, it looks like a constant, consider changing it to ALL_CAPITAL

File: app/routers/agenda.py [hunk 7]
```diff
@@ -39,6 +40,21 @@ def agenda(
...
+    start_date_graph, end_date_graph = calc_dates_range_for_agenda(
+        0, 0, get_a_week_int)
```
Comment (yammesicka) [inline app/routers/agenda.py]: Consider changing to keyword arguments to make it more readable

File: app/routers/agenda.py (hunk 6 above)
Comment (yammesicka) [inline app/routers/agenda.py]: Move all the logic to another function, preferably in a different file.

File: app/routers/agenda.py [hunk 8]
```diff
@@ -39,6 +40,21 @@ def agenda(
...
+
+    events_this_week = agenda_events.get_events_per_dates(
+        db, user_id, start_date_graph, end_date_graph
+    )
+    events_for_graph = {
+        str(start_date_graph +
+            timedelta(i)): 0 for i in range(get_a_week_int + 1)}
+
+    for event_obj in events_this_week:
+        events_for_graph[str(event_obj.start.date())] += 1
```
Comment (yammesicka) [inline app/routers/agenda.py]: Prefer splitting this to two lines for the sake of readability

File: app/routers/agenda.py [hunk 9]
```diff
@@ -39,6 +40,21 @@ def agenda(
...
+
+    get_a_week_int = 6
+    start_date_graph, end_date_graph = calc_dates_range_for_agenda(
+        0, 0, get_a_week_int)
+
+    events_this_week = agenda_events.get_events_per_dates(
+        db, user_id, start_date_graph, end_date_graph
+    )
+    events_for_graph = {
+        str(start_date_graph +
```
Comment (yammesicka) [inline app/routers/agenda.py]: I'm sure we can make these three lines bit more readable :) Please fix the line splitting

File: app/static/graph.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    const ctx = document.getElementById("myChart");
+    ctx.style.backgroundColor = "rgba(255, 255, 255, 1)";
+    const myChart = new Chart(ctx, {
+        type: "bar",
+        data: {
+            labels: Object.keys(events),
+            datasets: [{
+                label: "# Events",
+                data: Object.values(events),
+                backgroundColor: [
```
Comment (yammesicka) [inline app/static/graph.js:15]: I'd suggest on exporting this one to a `const` outside the function

File: app/templates/agenda.html [hunk 11]
```diff
@@ -25,28 +19,44 @@
...
-    {% endfor %}
+        {% endfor %} {% endfor %}
+    </div>
+
+
+    <div class="col-4">
+        <div class="card-body">
+            <canvas id="myChart" width="400" height="400"></canvas>
+            <div class="container" id="graph">
+                <script src="https://cdn.jsdelivr.net/npm/chart.js@2.8.0"></script>
```
Comment (yammesicka) [inline app/templates/agenda.html]: Move to the end of the `<body>` block

**Review by yammesicka — COMMENTED** [review]
Awesome! I've added few of my insights. Please take a look :)

File: app/routers/agenda.py (hunk 6 above)
Comment (nadav-pesach) [reply app/routers/agenda.py:44]: About the logic here, its doing exactly what the rout does, its part of the function.

Get start end day and return a list for a week.
IT has to be the same function has the rout,  i use all the rout parm to get this and i have to return it has part of the page..
 
It is the same as when you search for date it is the same logic

Waiting for your answer🙏

File: app/static/graph.js (hunk 10 above)
Comment (nadav-pesach) [reply app/static/graph.js:15]: I changed it don't know why it isn't "outdated",
image

File: app/routers/agenda.py (hunk 6 above)
Comment (nadav-pesach) [reply app/routers/agenda.py]: done

File: app/routers/agenda.py (hunk 7 above)
Comment (nadav-pesach) [reply app/routers/agenda.py]: done

File: app/routers/agenda.py (hunk 6 above)
Comment (nadav-pesach) [reply app/routers/agenda.py]: done

File: app/routers/agenda.py (hunk 8 above)
Comment (nadav-pesach) [reply app/routers/agenda.py]: done

File: app/routers/agenda.py (hunk 9 above)
Comment (nadav-pesach) [reply app/routers/agenda.py]: done

File: app/templates/agenda.html (hunk 11 above)
Comment (nadav-pesach) [reply app/templates/agenda.html]: done

File: app/internal/agenda_events.py [hunk 12]
```diff
@@ -89,3 +90,22 @@ def get_events_in_time_frame(
     """Yields all user's events in a time frame."""
     events = get_all_user_events(db, user_id)
     yield from filter_dates(events, start_date, end_date)
+
+
+def make_dict_for_graph_data(db: Session, user_id: int) -> Dict:
```
Comment (yammesicka) [inline app/internal/agenda_events.py]: Specify what types the `Dict` contains: `Dict[key, value]`

File: app/templates/agenda.html [hunk 13]
```diff
@@ -60,4 +69,5 @@ <h1 id="dates">{{ start_date.strftime("%d/%m/%Y") }} - {{ end_date.strftime("%d/
     {% endfor %}
   </div>
 
+</div>
```
Comment (yammesicka) [inline app/templates/agenda.html]: You sure you need to close this div here?

File: app/templates/base.html [hunk 14]
```diff
@@ -75,6 +75,8 @@
     <!-- This bootstrap version is needed here because of the toggler bug in the beta version-->
     <script src="https://stackpath.bootstrapcdn.com/bootstrap/5.0.0-alpha2/js/bootstrap.bundle.min.js" integrity="sha384-BOsAfwzjNJHrJ8cZidOg56tcQWfp6y72vEJ8xQ9w6Quywb24iOsW913URv1IS4GD" crossorigin="anonymous"></script>
     <script src="{{ url_for('static', path='/horoscope.js') }}"></script>
+    <script src="{{ url_for('static', path='/graph.js') }}"></script>
+    <script src="https://cdn.jsdelivr.net/npm/chart.js@2.8.0"></script>
```
Comment (yammesicka) [inline app/templates/base.html]: We already have Chart.js 2.9.4 included, see few lines above

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/templates/base.html (hunk 14 above)
Comment (nadav-pesach) [reply app/templates/base.html]: done

File: app/templates/agenda.html (hunk 13 above)
Comment (nadav-pesach) [reply app/templates/agenda.html]: done

File: app/internal/agenda_events.py (hunk 12 above)
Comment (nadav-pesach) [reply app/internal/agenda_events.py]: done

File: app/internal/agenda_events.py [hunk 15]
```diff
@@ -89,3 +90,22 @@ def get_events_in_time_frame(
     """Yields all user's events in a time frame."""
     events = get_all_user_events(db, user_id)
     yield from filter_dates(events, start_date, end_date)
+
+
+def make_dict_for_graph_data(db: Session, user_id: int) -> Dict[str, int]:
+    """create a dict with number of events per day for current week"""
+    WEEK_DAYS = 6
+    start_date, end_date = agenda.calc_dates_range_for_agenda(
+        0, 0, WEEK_DAYS)
```
Comment (yammesicka) [inline app/internal/agenda_events.py]: add parameter names to make the call more understandable: X=0, Y=0, ..

File: app/internal/agenda_events.py [hunk 16]
```diff
@@ -89,3 +90,22 @@ def get_events_in_time_frame(
...
+
+    events_this_week = get_events_per_dates(
+        db, user_id, start_date, end_date
+    )
+    events_for_graph = {
+        str(start_date + timedelta(i)): 0 for i in range(WEEK_DAYS + 1)
+    }
```
Comment (yammesicka) [inline app/internal/agenda_events.py]: This should be on a separate function

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Comment by yammesicka** [comment]
Probably the last set of fixes :)

File: app/internal/agenda_events.py (hunk 16 above)
Comment (nadav-pesach) [reply app/internal/agenda_events.py:106]: this is the separate function.. this was the part i separated from the route.

File: app/internal/agenda_events.py (hunk 15 above)
Comment (nadav-pesach) [reply app/internal/agenda_events.py]: done

Comment (nadav-pesach) [reply app/internal/agenda_events.py]: I actually changed it to make it more readable

File: app/internal/agenda_events.py (hunk 16 above)
Comment (nadav-pesach) [reply app/internal/agenda_events.py]: done

File: app/internal/agenda_events.py (hunk 15 above)
Comment (nadav-pesach) [reply app/internal/agenda_events.py]: done

File: app/internal/agenda_events.py [hunk 17]
```diff
@@ -89,3 +89,28 @@ def get_events_in_time_frame(
...
+                            ) -> Tuple[Union[Iterator[Event], list], Dict]:
+    WEEK_DAYS = 7
+    start_date = date.today()
+    end_date = start_date + timedelta(days=WEEK_DAYS - 1)
+
+    events_this_week = get_events_per_dates(
+        db, user_id, start_date, end_date
+    )
+    events_for_graph = {
+        str(start_date + timedelta(i)): 0 for i in range(WEEK_DAYS)
```
Comment (ivarshav) [inline app/internal/agenda_events.py:104]: for _ in ..

Comment (nadav-pesach) [reply app/internal/agenda_events.py:104]: hey, I'm using the i in timedelta(i)...

**Review by yammesicka — APPROVED** [review]

**Comment by yammesicka** [comment]
Resolve conflicts and send it to me for merge

**Pull request merged**